authors/apps/authentication/tests_/test_endpoints.py

```diff
@@ -0,0 +1,65 @@
+from django.test import TestCase
+from rest_framework.test import  APIClient
+from rest_framework import status
+from django.urls import reverse
+import json
+from authors.apps.authentication.models import User
+# from authors.apps.authentication import
+
+class TestUserEndPoints(TestCase):
+    def setUp(self):
+        self.user_data={
+            "user":{
+                "username": "Jacob",
+                "email": "jake@jake.jake",
+                "password": "jakejake"
+            }
+            }
+        self.verified_user=User.objects.create_user(username='testuser',email='testemail@test.com',password='testpassword')
+        
+        self.verified_user_login_credentials={
+                    "user":{
+                        "email": "testemail@test.com",
+                        "password": "testpassword"
+                    }
+                    }
+    
+        self.invalid_user={
+                "user":{
+                    "email":"adddd@gmail.com",
+                    "password":"Aaddadasas"
+                  
+                }
+                }
+        
+        self.client=APIClient()
+
+    def test_user_regsitration(self):
+        resp=self.client.post(reverse('authentication:signup'),content_type='application/json',data=json.dumps(self.user_data))
+        self.assertEqual(resp.status_code,status.HTTP_201_CREATED)
+        self.assertIn('Jacob',str(resp.data))
+   
+   
+    
+    def test_login_user(self):
+         resp=self.client.post(reverse('authentication:login'),content_type='application/json',data=json.dumps(self.verified_user_login_credentials))
+         self.assertEqual(resp.status_code,200)
+
+
+    def test_login_with_invalid_user(self):
+        expected_response={
+            "errors": {
+                "error": [
+                    "A user with this email and password was not found."
+                ]
+            }
+            }
```

Comment:
This looks good but this indentation may cause some future issues as we make more tests. Maybe look into it.

Reply:
sorry forgot to install pep8